### PR TITLE
fix: Allow platform dependencies to be linked

### PR DESCRIPTION
**Pull request by Artur-**
Do not fail if a managed version is linked to a local copy and thus the version cannot be parsed

### CODE REVIEW

File: flow-server/src/main/java/com/vaadin/flow/server/frontend/TaskUpdatePackages.java [hunk 1]
```diff
@@ -327,9 +327,18 @@ protected static boolean pinPlatformDependency(JsonObject packageJson,
         // packages exist at this point
         assert vaadinDeps != null : "vaadin{ dependencies { } } should exist";
         assert packageJsonDeps != null : "dependencies { } should exist";
+
+        FrontendVersion packageJsonVersion;
+        try {
+            packageJsonVersion = new FrontendVersion(
+                    packageJsonDeps.getString(pkg));
+        } catch (Exception e) {
```
Comment (mshabarov) [inline flow-server/src/main/java/com/vaadin/flow/server/frontend/TaskUpdatePackages.java:335]: Would it be more correct to scope the caught exception to only `NumberFormatException` ?

**Review by mshabarov — CHANGES_REQUESTED** [review]
Also would need a unit test.

File: flow-server/src/main/java/com/vaadin/flow/server/frontend/TaskUpdatePackages.java (hunk 1 above)
Comment (Artur-) [reply flow-server/src/main/java/com/vaadin/flow/server/frontend/TaskUpdatePackages.java:335]: What should we do on another exception?

Comment (mshabarov) [reply flow-server/src/main/java/com/vaadin/flow/server/frontend/TaskUpdatePackages.java:335]: We don't expect other exceptions in the `FrontendVersion` CTOR, if it throws NPE or other runtime exception, then it throws.

**Review by mshabarov — APPROVED** [review]

**Pull request merged**